### PR TITLE
The attempt to optimise merge was incorrect

### PR DESCRIPTION
**Pull request by russelldb**
If LHS vclock dominates RHS vclock, there can still be removes
that need to be observed, so that optimisation was bogus.

### CODE REVIEW

**Comment by russelldb** [comment]
Hat tip to https://github.com/asonge for the find, and the test.

File: src/riak_dt_orswot.erl [hunk 1]
```diff
@@ -172,41 +172,25 @@ remove_all([Elem | Rest], Actor, ORSet) ->
...
+
+    {Clock, Entries}.
+
+
 
-%% @private check if either clock dominates the other
--spec either_dominates(riak_dt_vclock:vclock(), riak_dt_vclock:vclock()) ->
-                              riak_dt_vclock:vclock() | concurrent.
-either_dominates(LHSClock, RHSClock) ->
-    case {riak_dt_vclock:descends(LHSClock, RHSClock),
```
Comment (lenary) [inline src/riak_dt_orswot.erl:201]: Is the optimisation worth keeping if you just change these lines to strict_descends? Frankly, this is what I always thought `dominates` meant...

Comment (lenary) [inline src/riak_dt_orswot.erl:201]: Oh, because there is no strict_descends... d'oh.

Comment (russelldb) [inline src/riak_dt_orswot.erl:201]: No, there is a dominates in riak_dt_vclock, but even that is not right. Even if LHS dominates RHS, you still need to compare the contents against RHS's clock.

**Comment by russelldb** [comment]
Gonna YOLO merge as develop is broken, and this is essentially a rollback.

**Pull request merged**